### PR TITLE
Map syphilis history to FHIR

### LINKED ISSUE #7564: [Syphilis] FHIR Mapping for Syphilis AOE data
## Why should we do this?
- In order to report Syphilis data to STLTs we need to map the disease specific AOE questions to FHIR

## User story
As a SimpleReport engineer, I want syphilis AOE data sent in the appropriate FHIR format, so the data can flow through ReportStream and be available to STLTs

## Action Requested
- Update FHIR converter logic to include syphilis results and AOE responses in FHIR bundle

## Acceptance Criteria

**Review**
-   [ ] Reviewed with design and/or design lead
-   [ ] Reviewed with content strategist for final approval
-   [ ] Reviewed with product and engineering

## Additional Context
Syphilis AOE questions are listed in #7436 and are currently subject to change (4/16/2024).  Once they are finalized/confirmed we will need to coordinate with ReportStream to verify the appropriate FHIR format.

--
From Nicole Gallaway, CDC SME for Syphilis:

> _There is no ‘verified list of symptoms’. I think the signs/symptom list you show looks generally good, except alopecia appears missing._ 

Suggested order:

```
If Syphilis result [Radio button] Positive

What is the gender of their sexual partners? (Select all that apply)
[checkbox] Female
[checkbox] Male
[checkbox] Transwoman
[checkbox] Transman
[checkbox] Nonbinary or gender non-conforming
[checkbox] Gender identity not listed here
[checkbox] Prefer not to answer

Has the patient been told they have syphilis before?
[Radio button] Yes: 1087151000119108
[Radio button] No: 14732006
[Radio button] Prefer not to answer: 261665006

Is the patient pregnant?
[Radio button] Yes
[Radio button] No
[Radio button] Prefer not to say

Is the patient currently experiencing or showing signs of symptoms?
[Radio button] Yes
[Radio button] No

Which symptoms are they experiencing?
- Genital sore/lesion (SNOMED: 724386005)
- Anal sore/lesion (SNOMED: 195469007)
- Sore(s) in mouth/lips (SNOMED: 26284000)
- Body Rash (SNOMED: 266128007)
- Palmar (hand)/plantar (foot) rash (SNOMED: 56940005)
- Flat white warts (SNOMED: 91554004)
- Hearing loss (SNOMED: 15188001)
- Blurred vision (SNOMED: 246636008)
- Alopecia (SNOMED: 56317004)

Date of symptom onset
[MM/DD/YYYY]

[Button] Submit results
```

### PR DESCRIPTION
**Pull request by emyl3**
# BACKEND PULL REQUEST

## Related Issue

- Resolves #7564 

## Changes Proposed

- maps syphilis history to FHIR
- bulk upload results
  - adds validation to the `syphilis_history` column so that it is [one of these values](https://github.com/CDCgov/prime-simplereport/blob/923e291b3aeff3a67652f8c3c181cb42cff1bf0f/backend/src/main/java/gov/cdc/usds/simplereport/validators/CsvValidatorUtils.java#L185)

## Additional Information
- [see context here](https://skylight-hq.slack.com/archives/C0411VC78DN/p1727372370949959?thread_ts=1721667117.540449&cid=C0411VC78DN) about how the shape of the FHIR message for this observation 

## Testing
- Use this branch `elisa/with-log-8100-fhir-syphilis-aoe` to test uploading syphilis results or entering syphilis results individually
  - search in your backend console `ZIPZAP` to see the shape of the msg

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java [hunk 1]
```diff
@@ -1629,7 +1659,7 @@ public Bundle createFhirBundle(ConditionAgnosticCreateFhirBundleProps props) {
 
     var bundle =
         new Bundle()
-            .setType(Bundle.BundleType.MESSAGE)
+            .setType(BundleType.MESSAGE)
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/converter/FhirConverter.java:1662]: ⚠️ running `./gradlew spotlessApply` removes this and the one on line `1671`

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java [hunk 2]
```diff
@@ -624,6 +624,7 @@ public List<FeedbackMessage> validateIndividualValues() {
               testResult,
               DiseaseService.SYPHILIS_NAME,
               List.of(gendersOfSexualPartners, pregnant, syphilisHistory, symptomaticForDisease)));
+      errors.addAll(validateYesNoAnswer(syphilisHistory));
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java:627]: ⚠️ added validation here for the `syphilis_history` column

File: backend/src/test/resources/testResultUpload/test-results-upload-all-fields.csv [hunk 3]
```diff
@@ -1,3 +1,3 @@
...
...t2,patient_city,patient_state,patient_zip_code,patient_county,patient_phone_number,patient_dob,patient_gender,patient_race,patient_ethnicity,patient_preferred_language,patient_email,accession_number,equipment_model_name,test_performed_code,test_result,order_test_date,specimen_collection_date,testing_lab_specimen_received_date,test_result_date,date_result_released,specimen_type,ordering_provider_id,ordering_provider_last_name,ordering_provider_first_name,ordering_provider_middle_name,ordering_provider_street,ordering_provider_street2,ordering_provider_city,ordering_provider_state,ordering_provider_zip_code,ordering_provider_phone_number,testing_lab_clia,testing_lab_name,testing_lab_street,testing_lab_street2,testing_lab_city,testing_lab_state,testing_lab_zip_code,testing_lab_phone_number,pregnant,employed_in_healthcare,symptomatic_for_disease,illness_onset_date,resident_congregate_setting,residence_type,hospitalized,icu,ordering_facility_name,ordering_facility_street,ordering_facility_street2,ordering_facility_city,ordering_facility_state,ordering_facility_zip_code,ordering_facility_phone_number,comment,test_result_status,test_ordered_code,genders_of_sexual_partners,syphilis_history
```
Comment (emyl3) [inline backend/src/test/resources/testResultUpload/test-results-upload-all-fields.csv:1]: ⚠️ change here is adding a `syphilis_history` column that is empty for each of the two rows

File: backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java [hunk 4]
```diff
@@ -618,6 +618,7 @@ public List<FeedbackMessage> validateIndividualValues() {
               testResult, DiseaseService.HIV_NAME, List.of(gendersOfSexualPartners, pregnant)));
     }
 
+    errors.addAll(validateYesNoAnswer(syphilisHistory));
```
Comment (emyl3) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java:621]: ⚠️ added validation here for the `syphilis_history` column

**Review by bobbywells52 — APPROVED** [review]
🚀 Very clean -- thanks for your work on this!

**Review by mpbrown — APPROVED** [review]
LGTM! Nice work on this!

**Pull request merged**